### PR TITLE
chore: upgrade kube-state-metrics to 2.14.0

### PR DESCRIPTION
**Pull request by echlebek**
### Checklist

- [x] Changelog updated or skip changelog label added
- [x] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

**Comment by ccressent** [comment]
The upgrade looks fine, but have you checked that there are no important or even breaking changes since 2.7? I know it's "only" minor versions, but you'd be surprised how many components use very loose definitions of semver 😅

**Comment by echlebek** [comment]
@ccressent Looks like all did not go well, but it's because our chart is pulling kube-state-metrics out of our own ECR. I have no idea why it's doing that but it seems related to some openshift incompatibility.

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -1098,7 +1098,7 @@ kube-prometheus-stack:
     ## so we bump the image manually.
     image:
       repository: public.ecr.aws/sumologic/kube-state-metrics
-      tag: "v2.7.0"
+      tag: "v2.14.0"
```
[the remark this answers is not in the document]
Comment (echlebek) [reply deploy/helm/sumologic/values.yaml:1101]: @rnishtala-sumo I can do that once I get the integration tests passing

[the remark this answers is not in the document]
Comment (rnishtala-sumo) [reply deploy/helm/sumologic/values.yaml:1101]: Tested this, LGTM!

**Review by rnishtala-sumo — APPROVED** [review]

**Pull request merged**